src/wallet/rpcnames.cpp

```diff
@@ -489,7 +489,12 @@ name_firstupdate (const JSONRPCRequest& request)
 
   const uint256 prevTxid = ParseHashV (request.params[2], "txid");
 
-  const valtype value = DecodeValueFromRPCOrThrow (request.params[3], options);
+  const bool isDefaultVal = (request.params.size () < 4 || request.params[3].isNull ());
+  const std::string& defaultValue = "";
```

Comment:
I think this is not correct, it should be `const std::string` (without being a reference).  But I would even go further, and just remove this line, and use `valtype ()` below.  I think the code is clear enough that way.